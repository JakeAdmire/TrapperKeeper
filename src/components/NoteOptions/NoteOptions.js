import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types'

const NoteOptions = ({ handleType, deleteNote, handleClose, type }) => {

  return (
    <div className="Note-Options">
      <button className="Note-Save" onClick={handleType} type="submit">
        <i className="fas fa-save"></i>
      </button>
      {
        type === "existing-note" &&
          <button className="Note-Delete" onClick={deleteNote}>
            <i className="fas fa-trash-alt"></i>
          </button>
      }
      <NavLink onClick={handleClose} to="/" className="Note-Close">
        <button>Close</button>
      </NavLink>
    </div>
  )
}

NoteOptions.propTypes = {
  handleType: PropTypes.func.isRequired,
  deleteNote: PropTypes.func.isRequired,
  handleClose: PropTypes.func.isRequired
}

export default NoteOptions;