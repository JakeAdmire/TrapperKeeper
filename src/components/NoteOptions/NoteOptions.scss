.Note-Options {
  display: flex;
  justify-content: flex-end;
  padding: .5em;
  margin-top: 1em;
  align-items: baseline;

  button {
    background: none;
    outline: none;
    border: none;
    font-size: 1.2em;
    cursor: pointer;
    color: #6f6f6f;
    opacity: 1;
    transition: all .3s ease;
    margin-right: .5em;

    &:hover {
      opacity: .6;
    }
  }

  .Note-Save {
    i {
      position: relative;
      bottom: -3px;
      color: #8a846d;
    }
  }

  .Note-Delete {
    i {
      position: relative;
      bottom: -2px;
      color: #8a846d;
    }
  }

  .Note-Close {

    button {
      margin-right: 0;
      font-family: $font-Roboto;
      outline: none;
      cursor: pointer;
      border: none;
      height: 100%;
      box-sizing: border-box;
      color: $baseYellow;
      background-color: #8a846d;
      overflow: hidden;
      text-overflow: ellipsis;
      white-space: nowrap;
      letter-spacing: .01785714em;
      font-size: .75rem;
      font-weight: 500;
      line-height: .6;
      height: 24px;
      padding: 8px 15px;
      border-radius: 4px;
    }
  }


}
