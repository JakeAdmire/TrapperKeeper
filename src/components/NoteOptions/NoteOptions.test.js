import React from 'react';
import { shallow } from 'enzyme';
import NoteOptions from './NoteOptions';

describe('NoteOptions', () => {
  let wrapper;

  const mockHandleType = jest.fn()
  const mockDeleteNote = jest.fn()
  const mockHandleClose = jest.fn()

  beforeEach(() => {
    wrapper = shallow(<NoteOptions handleType={mockHandleType} deleteNote={mockDeleteNote} handleClose={mockHandleClose} />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should call handleType when the save button is clicked', () => {
    wrapper.find('.Note-Save').simulate('click')
    expect(mockHandleType).toHaveBeenCalled()
  })

  it('should call deleteNote when the delete button is clicked', () => {
    wrapper.find('.Note-Delete').simulate('click')
    expect(mockDeleteNote).toHaveBeenCalled()
  })

  it('should call handleClose when the close note button is clicked', () => {
    wrapper.find('.Note-Close').simulate('click')
    expect(mockHandleClose).toHaveBeenCalled()
  })
})