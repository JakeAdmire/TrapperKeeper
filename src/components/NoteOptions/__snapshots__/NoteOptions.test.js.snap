// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteOptions should match the snapshot 1`] = `
<div
  className="Note-Options"
>
  <button
    className="Note-Save"
    onClick={[MockFunction]}
    type="submit"
  >
    <i
      className="fas fa-save"
    />
  </button>
  <button
    className="Note-Delete"
    onClick={[MockFunction]}
  >
    <i
      className="fas fa-trash-alt"
    />
  </button>
  <NavLink
    className="Note-Close"
    onClick={[MockFunction]}
    to="/"
  >
    <button>
      Close
    </button>
  </NavLink>
</div>
`;
